src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurCashAdvancePdpFeedFileServiceImpl.java

```diff
@@ -1,36 +1,299 @@
+        pdpAccountingEntry.setObjectCd(getConcurParamterValue(ConcurParameterConstants.DEFAULT_TRAVEL_REQUEST_OBJECT_CODE));
+        pdpAccountingEntry.setSubObjectCd(detailFileLine.getSubObjectCode());
+        pdpAccountingEntry.setOrgRefId(detailFileLine.getOrgRefId());
+        pdpAccountingEntry.setProjectCd(detailFileLine.getProjectCode());
+        pdpAccountingEntry.setAmount(detailFileLine.getRequestAmount());
+        return pdpAccountingEntry;
+    }
 
-    public ConcurCashAdvancePdpFeedFileServiceImpl() {
-        super();
+    private PdpFeedPayeeIdEntry buildPdpFeedPayeeIdEntry(ConcurRequestExtractRequestDetailFileLine detailFileLine) {
+        PdpFeedPayeeIdEntry payeeIdEntry = new PdpFeedPayeeIdEntry();
+        payeeIdEntry.setContent(detailFileLine.getEmployeeId());
+        if (StringUtils.equalsIgnoreCase(detailFileLine.getPayeeIdType(), ConcurRequestExtractPdpConstants.ValidationConstants.EMPLOYEE_INDICATOR)) {
+            payeeIdEntry.setIdType(ConcurConstants.EMPLOYEE_PAYEE_STATUS_TYPE_CODE);
+        } else if (StringUtils.equalsIgnoreCase(detailFileLine.getPayeeIdType(), ConcurRequestExtractPdpConstants.ValidationConstants.NON_EMPLOYEE_INDICATOR)) {
+            payeeIdEntry.setIdType(ConcurConstants.NON_EMPLOYEE_PAYEE_STATUS_TYPE_CODE);
+        } else {
+            LOG.error("Invalid PayeeIdType detected in buildPdpFeedPayeeIdEntry AFTER validation while building PDP output file:" + detailFileLine.getPayeeIdType());
+        }
+        return payeeIdEntry;
     }
 
-    @Override
-    public List<String> getRequiredDirectoryNames() {
-        List<String> directoryNames = new ArrayList<String>();
-        directoryNames.add(getOutgoingDirectoryName());
-        return directoryNames;
+    private PdpFeedHeaderEntry buildPdpFeedHeaderEntry(Date batchDate) {
+        PdpFeedHeaderEntry header = new PdpFeedHeaderEntry();
+        header.setChart(getConcurParamterValue(ConcurParameterConstants.CONCUR_CUSTOMER_PROFILE_LOCATION));
+        header.setCreationDate(formatDate(batchDate));
+        header.setSubUnit(getConcurParamterValue(ConcurParameterConstants.CONCUR_CUSTOMER_PROFILE_SUB_UNIT));
+        header.setUnit(getConcurParamterValue(ConcurParameterConstants.CONCUR_CUSTOMER_PROFILE_UNIT));
+        return header;
     }
 
-    public void createDoneFileFor(String concurCashAdvancePdpFeedFileName) {
-        LOG.debug("Done file creation was requested.");
+    private PdpFeedTrailerEntry buildPdpFeedTrailerEntry(int totalDetailRecordCount, KualiDecimal totalDetailRecordAmount) {
+        PdpFeedTrailerEntry trailer = new PdpFeedTrailerEntry();
+        trailer.setDetailCount(totalDetailRecordCount);
+        trailer.setDetailTotAmt(totalDetailRecordAmount);
+        return trailer;
     }
 
-    public void setOutgoingDirectoryName(String outgoingDirectoryName) {
-        this.outgoingDirectoryName = outgoingDirectoryName;
+    //common utility method
```

Comment:
I spoke with Nancy, and creating ConcurBatchUtilityServiceImpl seems to make the most sense.